website_src/docs/r/service_dictionary_items_v1.html.markdown.tmpl

```diff
@@ -133,15 +140,61 @@ resource "fastly_service_v1" "myservice" {
 
 // This resource is dynamically creating the items from the local variables through for expressions and functions.
 resource "fastly_service_dictionary_items_v1" "project" {
+  for_each = {
+    for d in fastly_service_v1.myservice.dictionary : d.name => d if d.name == local.dictionary_name
+  }
   service_id = fastly_service_v1.myservice.id
-  dictionary_id = {for d in fastly_service_v1.myservice.dictionary : d.name => d.dictionary_id}[local.dictionary_name]
+  dictionary_id = each.value.dictionary_id
   items = {
     for division in local.host_divisions:
       division => format("%s.%s", division, local.host_base)
   }
 }
 ```
 
+## Example Usage (Terraform >= 0.12.0 && < 0.12.6)
+
+`for_each` attributes were not available in Terraform before 0.12.6, however, users can still use `for` expressions to achieve
+similar behaviour as seen in the example below.
+
+~> **Warning:** Terraform might not properly calculate implicit dependencies on computed attributes when using `for` expressions
+
+For scenarios such as adding a Dictionary to a service and at the same time, creating the Dictionary entries (`fastly_service_dictionary_items_v1`)
+resource, Terraform will not calculate compute implicit dependencies correctly on `for` expressions. This will result in index lookup
```

Comment:
I think this was a typo as "calculate compute" sounds weird, but it may have been intentional? I went with "calculate" to be consistent with the warning sentence above. 

```suggestion
resource, Terraform will not calculate implicit dependencies correctly on `for` expressions. This will result in index lookup
```